### ISSUE TITLE
[Bug]: IINA won't open: giving javascript error

### ISSUE THREAD

**Issue by gradygambrell** [FreeTubeApp/FreeTube#4132]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

1. Go to a video of choice
2. Click on the video and let it play
3. Click on the icon that allows to use another player
4. When error occurs, I:
5. Go into settings and click on the External Player section
6. Click on "Ignore unsupported action warnings"
7. type the full path of IINA player in the custom player (IINA is in my Applications folder)
8. try again with no prevail.

### Expected Behavior

To be able to open the IINA player.

### Issue Labels

feature stopped working, text/string issue, usability issue, visual bug

### FreeTube Version

v0.19.1 and nightly 3525

### Operating System Version

macOS Sonoma 14.0

### Installation Method

.dmg

### Primary API used

Invidious API

### Additional Information

I check the issue here, but hasn't be responded since Sept 5th (unsure if it's this year or 2022):

https://github.com/FreeTubeApp/FreeTube/issues/2158

The thing is closed and is unsure if an solution have been made.

I have used the nightly build, with no success.

### Nightly Build

- [X] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by ChunkyProgrammer** [comment]
have you added the directory of your IINA executable to your path environment variable?

**Comment by gradygambrell** [comment]
> have you added the directory of your IINA executable to your path environment variable?

I have tried that, still gave me the Javascript error:

```
Uncaught Exception:
Error: spawn /Applications/IINA.app/Contents/MacOS/IINA  ENOENT
at ChildProcess._handle.onexit (node:internal/child_process:283:19)
at onErrorNT (node:internal/child_process:478:16)
at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
```

**Comment by Adecom16** [comment]
Can you assign this to me and receive an instant PR

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@trostboot any thoughts one this?

**Comment by trostboot** [comment]
I don't have any Apple devices so can't really debug this. Does spawning external players work at all on MacOS the way it's currently implemented? @gradygambrell test it with e.g. mpv.

**Comment by gradygambrell** [comment]
> I don't have any Apple devices so can't really debug this. Does spawning external players work at all on MacOS the way it's currently implemented? @gradygambrell test it with e.g. mpv.

Tried MPV without any path name (default folder is /Applications/mpv.app and gives me this JS error:

```
Uncaught Exception:
Error: spawn mpv ENOENT
at ChildProcess._handle.onexit (node:internal/child_process:283:19)
at onErrorNT (node:internal/child_process:478:16)
at process.processTicksAndRejections (node:internal/process/task_queues:83:21)

```

Tried with this path name: /Applications/mpv.app/Contents/MacOS/mpv-bundle and it opens the application, but then closes right away.

Tried VLC player using /Applications/VLC.app and another JS error appeared.

```
Uncaught Exception:
Error: spawn vlc ENOENT
at ChildProcess._handle.onexit (node:internal/child_process:283:19)
at onErrorNT (node:internal/child_process:478:16)
at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
```

HOWEVER,  using this path in the custom path: /Applications/VLC.app/Contents/MacOS/VLC, I was able to open it via VLC. Had to do an extra step by clicking on the title of the video to play.

TL;DR: mpv didn't work. VLC does, but with an special path.

**Comment by ChunkyProgrammer** [comment]
Could you send a screenshot of your external player configuration?

**Comment by gradygambrell** [comment]
> Could you send a screenshot of your external player configuration?

Yes.

![Screen Shot 2023-10-17 at 2 14 31 PM](https://github.com/FreeTubeApp/FreeTube/assets/97412521/e23bd5cc-4686-4a01-b3ed-e2ff44c874cb)


(This one is after everything is loaded)
![Screen Shot 2023-10-17 at 2 14 24 PM](https://github.com/FreeTubeApp/FreeTube/assets/97412521/f79cca79-8a72-41b5-9ca0-c16c7ef14c8b)

**Comment by vdbhb59** [comment]
This works fine for me on Windows as well as Linux. I usually have miniplayer, potplayer and inbuilt linux ones, and each one plays fine.
![image](https://github.com/FreeTubeApp/FreeTube/assets/60728004/784c2230-9514-483d-bc25-54a1451f9b75)

**Comment by sebma** [comment]
I have the same Javascript error. Can someone tell me what a IINA is ?

**Comment by superfreeman1989** [comment]
Same issue on macos

**Comment by avfx** [comment]
same issue on linux (fedora 39) using mpv, open a youtube video directly with mpv from the shell works, but not from freetube

**Comment by trostboot** [comment]
> same issue on linux (fedora 39) using mpv, open a youtube video directly with mpv from the shell works, but not from freetube

Hm, I can't repro this. Fresh Fedora 39 VM, updated to latest packages, only with the minimum additional packages needed for mpv and Freetube dev. Freetube default settings aside from external player -> mpv. Works from both the shell as well as from Freetube.
Could you give it a try with a fresh profile with default Freetube settings and see if it still breaks?

**Comment by avfx** [comment]
I did a Fresh Fedora 39 VM using VirtualBox, installed mpv , youtube-dl and freetube. All runs with default settings, mpv opens a YouTube video from the shell and Freetube still gives me the same error as I have in my host machine (see below). I originally thought it's because I have the flatpak version of mpv on my host, but it didn't seem to be the case as I didn't use flatpak on the VM.

![Screenshot from 2024-02-16 04-15-36](https://github.com/FreeTubeApp/FreeTube/assets/7225012/acee2697-0ddc-4563-a3eb-9f4173acdea1)

**Comment by avfx** [comment]
Thanks @trostboot. I found the issue, it's running the flatpak version of Freetube. It works when I use the rpm version.

**Comment by trostboot** [comment]
Okay well that makes more sense given the nature of Flatpaks. I have no idea if the MacOS version is similarly sandboxed as to cause the original issue, but either way, this is a bit out of my wheelhouse and not something I personally have the time to look deeper into.

**Comment by jurekcy** [comment]
I have the same issue with VLC on Win10. Fresh install of Freetube (I am new joiner)
![obraz](https://github.com/FreeTubeApp/FreeTube/assets/148444412/b174a06d-247f-4c85-b33b-c57532e46e21)

**Comment by superfreeman1989** [comment]
Issue still exists in latest release

**Comment by ngocphamm** [comment]
For me on macOS, putting `/opt/homebrew/bin/iina` into the `Custom External Player Executable` input works, as I installed iina with Homebrew.

**Comment by SOY75** [comment]
Hello
It works for me with /Applications/IINA.app/Contents/MacOS/IINA set as custom path.

**Comment by forgettablepyromaniac** [comment]
I'll throw my own hat in here and say that I am also coming across this issue as a new FreeTube user:
![image](https://github.com/user-attachments/assets/614161b6-0b29-4f08-bb93-4b2b96b8a2bb)

Clarifying the exact location of the exe doesn't help, either.

Right as I was going to upload this, though, I DID come across something small but substantial:
This error was only occurring on instances where there was a space in the path name.

"C:\Users\$USERNAME$\3D Objects\mpv\mpv.exe" would error, meanwhile "C:\Users\Terra\Desktop\mpv\mpv.exe" would not.

MPV still wouldn't open but I assume that's just something on my end from opening MPV where it didn't expect.

**Comment by superfreeman1989** [comment]
Not fixed

**Comment by exekutive** [comment]
> It works for me (without homebrew) with /Applications/IINA.app/Contents/MacOS/IINA set as custom path.

Would be better if Freetube did this automatically

**Comment by Dark-Thoughts** [comment]
Bummer. On Bazzite I'm kind of dependent on Flatpaks, assuming I don't want to fiddle with Distrobox even more, which is just such a hassle and prone to breakage.

**Comment by glottisfaun0000** [comment]
> > I don't have any Apple devices so can't really debug this. Does spawning external players work at all on MacOS the way it's currently implemented? [@gradygambrell](https://github.com/gradygambrell) test it with e.g. mpv.
> 
> Tried MPV without any path name (default folder is /Applications/mpv.app and gives me this JS error:
> 
> ```
> Uncaught Exception:
> Error: spawn mpv ENOENT
> at ChildProcess._handle.onexit (node:internal/child_process:283:19)
> at onErrorNT (node:internal/child_process:478:16)
> at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
> ```
> 
> Tried with this path name: /Applications/mpv.app/Contents/MacOS/mpv-bundle and it opens the application, but then closes right away.
> 
> Tried VLC player using /Applications/VLC.app and another JS error appeared.
> 
> ```
> Uncaught Exception:
> Error: spawn vlc ENOENT
> at ChildProcess._handle.onexit (node:internal/child_process:283:19)
> at onErrorNT (node:internal/child_process:478:16)
> at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
> ```
> 
> HOWEVER, using this path in the custom path: /Applications/VLC.app/Contents/MacOS/VLC, I was able to open it via VLC. Had to do an extra step by clicking on the title of the video to play.
> 
> TL;DR: mpv didn't work. VLC does, but with an special path.

So external players do not work on macOS?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
inquires regarding external players migrated to https://github.com/FreeTubeApp/FreeTube/discussions/7789
